js/ui/marker.js

```diff
@@ -5,22 +5,31 @@ module.exports = Marker;
 
 var DOM = require('../util/dom');
 var LngLat = require('../geo/lng_lat');
+var Point = require('point-geometry');
 
 /**
  * Creates a marker component
  * @class Marker
  * @param {HTMLElement=} element DOM element to use as a marker (creates a div element by default)
+ * @param {Object} options
+ * @param {Point} options.offset The offset in pixels to apply to this Marker relative to the element's top left corner.
```

Comment:
I think "offset" is preferable. Consider the style properties `icon-offset`, `text-offset`, versus `text-anchor` and [the `anchor` option of `Popup`](https://www.mapbox.com/mapbox-gl-js/api/#Popup).